diff --git a/DiligentCore/ThirdParty/glslang/SPIRV/spirv.hpp b/DiligentCore/ThirdParty/glslang/SPIRV/spirv.hpp
index afa89a58..2e548e02 100644
--- a/DiligentCore/ThirdParty/glslang/SPIRV/spirv.hpp
+++ b/DiligentCore/ThirdParty/glslang/SPIRV/spirv.hpp
@@ -47,6 +47,8 @@
 #ifndef spirv_HPP
 #define spirv_HPP
 
+#include <cstdint>
+
 namespace spv {
 
 typedef unsigned int Id;
@@ -1163,7 +1165,7 @@ enum Capability {
     CapabilityDotProduct = 6019,
     CapabilityDotProductKHR = 6019,
     CapabilityRayCullMaskKHR = 6020,
-    CapabilityCooperativeMatrixKHR = 6022,
+    CapabilityCooperativeMatrixKHR = 6022,
     CapabilityReplicatedCompositesEXT = 6024,
     CapabilityBitInstructions = 6025,
     CapabilityGroupNonUniformRotateKHR = 6026,
@@ -1693,9 +1695,9 @@ enum Op {
     OpCooperativeMatrixLoadKHR = 4457,
     OpCooperativeMatrixStoreKHR = 4458,
     OpCooperativeMatrixMulAddKHR = 4459,
-    OpCooperativeMatrixLengthKHR = 4460,
-    OpConstantCompositeReplicateEXT = 4461,
-    OpSpecConstantCompositeReplicateEXT = 4462,
+    OpCooperativeMatrixLengthKHR = 4460,
+    OpConstantCompositeReplicateEXT = 4461,
+    OpSpecConstantCompositeReplicateEXT = 4462,
     OpCompositeConstructReplicateEXT = 4463,
     OpTypeRayQueryKHR = 4472,
     OpRayQueryInitializeKHR = 4473,
@@ -2426,9 +2428,9 @@ inline void HasResultAndType(Op opcode, bool *hasResult, bool *hasResultType) {
     case OpCooperativeMatrixLoadKHR: *hasResult = true; *hasResultType = true; break;
     case OpCooperativeMatrixStoreKHR: *hasResult = false; *hasResultType = false; break;
     case OpCooperativeMatrixMulAddKHR: *hasResult = true; *hasResultType = true; break;
-    case OpCooperativeMatrixLengthKHR: *hasResult = true; *hasResultType = true; break;
-    case OpConstantCompositeReplicateEXT: *hasResult = true; *hasResultType = true; break;
-    case OpSpecConstantCompositeReplicateEXT: *hasResult = true; *hasResultType = true; break;
+    case OpCooperativeMatrixLengthKHR: *hasResult = true; *hasResultType = true; break;
+    case OpConstantCompositeReplicateEXT: *hasResult = true; *hasResultType = true; break;
+    case OpSpecConstantCompositeReplicateEXT: *hasResult = true; *hasResultType = true; break;
     case OpCompositeConstructReplicateEXT: *hasResult = true; *hasResultType = true; break;
     case OpTypeRayQueryKHR: *hasResult = true; *hasResultType = false; break;
     case OpRayQueryInitializeKHR: *hasResult = false; *hasResultType = false; break;
