diff --git a/source/opt/CMakeLists.txt b/source/opt/CMakeLists.txt
index 51153d7a..c0651e20 100644
--- a/source/opt/CMakeLists.txt
+++ b/source/opt/CMakeLists.txt
@@ -261,6 +261,13 @@ spvtools_pch(SPIRV_TOOLS_OPT_SOURCES pch_source_opt)
 
 add_library(SPIRV-Tools-opt ${SPIRV_TOOLS_LIBRARY_TYPE} ${SPIRV_TOOLS_OPT_SOURCES})
 
+if(MSVC)
+  set_source_files_properties(
+    fold_spec_constant_op_and_composite_pass.cpp
+    PROPERTIES COMPILE_OPTIONS "/GL-"
+  )
+endif()
+
 spvtools_default_compile_options(SPIRV-Tools-opt)
 target_include_directories(SPIRV-Tools-opt
   PUBLIC
