diff --git a/Inc/DirectXMath.h b/Inc/DirectXMath.h
index ae3cce1..d955f94 100644
--- a/Inc/DirectXMath.h
+++ b/Inc/DirectXMath.h
@@ -295,6 +295,8 @@ namespace DirectX
     constexpr uint32_t XM_CRMASK_CR6FALSE = 0x00000020;
     constexpr uint32_t XM_CRMASK_CR6BOUNDS = XM_CRMASK_CR6FALSE;
 
+    using size_t = std::size_t;
+
 #if defined(_M_ARM) || defined(_M_ARM64) || defined(_M_HYBRID_X86_ARM64) || defined(_M_ARM64EC) || __arm__ || __aarch64__
     constexpr size_t XM_CACHE_LINE_SIZE = 128;
 #else
